pkg/space/space.go

```diff
@@ -29,6 +32,42 @@ func NewSpace(userSignup *toolchainv1alpha1.UserSignup, targetClusterName string
 	return space
 }
 
+// NewSpaceWithFeatureToggles is the same as NewSpace() but also does a feature toggle lottery drawing
+// and adds the corresponding feature annotations for features which "won" and should be enabled for the space.
+func NewSpaceWithFeatureToggles(userSignup *toolchainv1alpha1.UserSignup, targetClusterName string, compliantUserName, tier string, config toolchainconfig.ToolchainConfig) *toolchainv1alpha1.Space {
+	s := NewSpace(userSignup, targetClusterName, compliantUserName, tier)
+	addFeatureToggles(s, config)
+	return s
+}
+
+// addFeatureToggles does "lottery" drawing for all feature toggles defined in the configuration according to their weights.
+// And it adds the corresponding feature annotation to the space for features that won and should be enabled for the space.
+func addFeatureToggles(space *toolchainv1alpha1.Space, config toolchainconfig.ToolchainConfig) {
+	toggles := config.Tiers().FeatureToggles()
+	var winners string
+	for _, t := range toggles {
+		weight := int(t.Weight())
+		// We generate a random number between 0 and 100. If the number is equal to or lower than the weight
+		// then the feature wins.
+		// We don't use recommended crypto/rand here because we don't need crypto grade random generator
+		// and math/rand with pseudo random numbers, which is much cheaper, is sufficient, so we are disabling the linter
+		if weight == 100 || (weight > 0 && rand.Intn(100) <= weight) { // nolint:gosec
+			// Winner!
+			if winners == "" {
+				winners = t.Name() // First winner so far
+			} else {
+				winners = fmt.Sprintf("%s,%s", winners, t.Name()) // Add to the comma separated list of existing winners
```

Comment:
or you could append each enabled feature to `var features = []string{}` and use `strings.Join(features, ",")` afterwards

Reply:
+1

Reply:
Yep. That would look better. Done in https://github.com/codeready-toolchain/host-operator/pull/1048/commits/44ecb0a8f2e403069dfdc5de71403e845cca05f5